--- a/src/apps/ociodisplay/main.cpp
+++ b/src/apps/ociodisplay/main.cpp
@@ -45,20 +45,9 @@
 namespace OIIO = OIIO_NAMESPACE;
 #endif
 
-#ifdef __APPLE__
-#include <OpenGL/gl.h>
-#include <OpenGL/glext.h>
-#include <GLUT/glut.h>
-#elif _WIN32
-#include <GL/glew.h>
-#include <GL/glut.h>
-#else
 #include <GL/glew.h>
 #include <GL/gl.h>
-#include <GL/glext.h>
 #include <GL/glut.h>
-#endif
-
 
 bool g_verbose = false;
 std::string g_filename;
