--- OpenColorIO-2.2.1.orig/share/cmake/modules/FindExtPackages.cmake.orig	2023-01-05 18:05:59.000000000 -0800
+++ OpenColorIO-2.2.1/share/cmake/modules/FindExtPackages.cmake	2023-04-30 23:37:03.723067360 -0700
@@ -178,7 +178,7 @@ if(OCIO_BUILD_PYTHON OR OCIO_BUILD_DOCS)
     endif()
 
     # Python
-    find_package(Python ${OCIO_PYTHON_VERSION} REQUIRED
+    find_package(Python ${OCIO_PYTHON_VERSION} EXACT REQUIRED   # oteodoro: use specific slot
                  COMPONENTS ${_Python_COMPONENTS})
 
     if(OCIO_BUILD_PYTHON)
