--- a/common/common.pro
+++ b/common/common.pro
@@ -1,7 +1,4 @@
 include (../general.pri)
-EXIF_DIR = ../external/jhead-2.95
-
-GLEWCODE = $$GLEWDIR/src/glew.c
 
 win32-msvc2005:DESTDIR = ../distrib
 win32-msvc2008:DESTDIR = ../distrib
@@ -43,9 +40,7 @@
 "
 INCLUDEPATH *= ../.. \
     $$VCGDIR \
-    $$EIGENDIR \
-    $$GLEWDIR/include \
-    $$EXIF_DIR
+    $$EIGENDIR
 TEMPLATE = lib
 
 linux:CONFIG += dll
@@ -69,7 +64,6 @@
 
 TARGET = common
 DEPENDPATH += .
-DEFINES += GLEW_STATIC
 
 win32-msvc:DEFINES += _CRT_SECURE_NO_WARNINGS
 
@@ -102,7 +96,6 @@
 			mlapplication.cpp \
 			scriptsyntax.cpp \
 			searcher.cpp \
-			$$GLEWCODE \
 			meshlabdocumentxml.cpp \
 			meshlabdocumentbundler.cpp \
 			ml_shared_data_context.cpp 
--- a/general.pri
+++ b/general.pri
@@ -7,7 +7,6 @@
 
 VCGDIR   = ../../../vcglib
 EIGENDIR = $$VCGDIR/eigenlib
-GLEWDIR = ../external/glew-2.1.0
 
 CONFIG += c++11
 
--- a/meshlab/meshlab.pro
+++ b/meshlab/meshlab.pro
@@ -1,7 +1,6 @@
 include (../general.pri)
 #CONFIG += debug_and_release
 DESTDIR = ../distrib
-EXIF_DIR = ../external/jhead-2.95
 
 DEFINES += QT_DISABLE_DEPRECATED_BEFORE=0x000000
 
@@ -9,9 +8,7 @@
 	.. \
 	../.. \
     $$VCGDIR \
-    $$EIGENDIR \
-    $$GLEWDIR/include \
-	$$EXIF_DIR
+    $$EIGENDIR
 DEPENDPATH += $$VCGDIR \
     $$VCGDIR/vcg \
     $$VCGDIR/wrap
@@ -154,7 +151,7 @@
 #	win32-g++:release:LIBS 			+= -L../common/release -lcommon
 #}
 
-linux:LIBS += -L$$PWD/../external/lib/linux -ljhead -L../distrib -lcommon -lGLU
+linux:LIBS += -L$$PWD/../external/lib/linux -ljhead -L../distrib -lcommon -lGLU -lGLEW
 linux:QMAKE_RPATHDIR += ../distrib
 
 # uncomment in your local copy only in emergency cases.
--- a/meshlabplugins/filter_func/filter_func.pro
+++ b/meshlabplugins/filter_func/filter_func.pro
@@ -8,7 +8,6 @@
 
 
 DEFINES += _UNICODE
-INCLUDEPATH += ../../external/muparser_v225/include
 
 # Note: we need static libs so when building muparser lib use 
 # ./configure --enable-shared=no
@@ -22,7 +21,4 @@
 win32-msvc2015:LIBS += ../../external/lib/win32-msvc2015/muparser.lib
 
 macx:LIBS            += $$MACLIBDIR/libmuparser.a
-linux-g++:LIBS       += $$PWD/../../external/lib/linux-g++/libmuparser.a
-linux-g++-32:LIBS       += $$PWD/../../external/lib/linux-g++-32/libmuparser.a
-linux-g++-64:LIBS       += $$PWD/../../external/lib/linux-g++-64/libmuparser.a
 win32-g++:LIBS		 += ../../external/lib/win32-gcc/libmuparser.a
--- a/meshlabplugins/io_3ds/io_3ds.pro
+++ b/meshlabplugins/io_3ds/io_3ds.pro
@@ -10,11 +10,9 @@
 				
 SOURCES       += 	meshio.cpp \
 			
-DEFINES += LIB3DS_STATIC
 
 TARGET        = io_3ds
 
-INCLUDEPATH += ../../external/lib3ds-1.3.0/
 
 # Notes on the paths of lib3ds files.
 # Once you have compiled the library put the lib files in a dir named lib/XXX/ 
@@ -26,9 +24,6 @@
 win32-msvc2005:LIBS	+= ../../external/lib/win32-msvc2005/3ds.lib
 win32-msvc2008:LIBS	+= ../../external/lib/win32-msvc2008/3ds.lib
 win32-g++:LIBS		+= ../../external/lib/win32-gcc/lib3ds.a
-linux-g++:LIBS += $$PWD/../../external/lib/linux-g++/lib3ds.a
-linux-g++-32:LIBS += $$PWD/../../external/lib/linux-g++-32/lib3ds.a
-linux-g++-64:LIBS += $$PWD/../../external/lib/linux-g++-64/lib3ds.a
 
 # unix:LIBS		+= -L../../../../code/lib/lib3ds-1.3.0/lib3ds/lib/unix -l3ds
 
--- a/meshlabplugins/io_ctm/io_ctm.pro
+++ b/meshlabplugins/io_ctm/io_ctm.pro
@@ -3,7 +3,6 @@
     $$VCGDIR/wrap/io_trimesh/import_ctm.h \
     $$VCGDIR/wrap/io_trimesh/export_ctm.h
 SOURCES += io_ctm.cpp
-INCLUDEPATH += ../../external/OpenCTM-1.0.3/lib
 TARGET = io_ctm
 
 win32-msvc.net:QMAKE_CXXFLAGS += /DOPENCTM_STATIC
@@ -21,8 +20,5 @@
 win32-msvc2012:LIBS	+= ../../external/lib/win32-msvc2012/openctm.lib
 win32-msvc2015:LIBS	+= ../../external/lib/win32-msvc2015/openctm.lib
 win32-g++:LIBS		+= ../../external/lib/win32-gcc/libopenctm.a
-linux-g++:LIBS		+= $$PWD/../../external/lib/linux-g++/libopenctm.a
-linux-g++-32:LIBS	+= $$PWD/../../external/lib/linux-g++-32/libopenctm.a
-linux-g++-64:LIBS	+= $$PWD/../../external/lib/linux-g++-64/libopenctm.a
 #macx:LIBS += ../../external/lib/macx/libopenctm.a
 macx:LIBS += $$MACLIBDIR/libopenctm.a
--- a/meshlabserver/meshlabserver.pro
+++ b/meshlabserver/meshlabserver.pro
@@ -28,8 +28,8 @@
 win32-msvc2010:  LIBS += -L../distrib -lcommon -lopengl32 -lGLU32
 win32-msvc2012:  LIBS += -L../distrib -lcommon -lopengl32 -lGLU32
 win32-msvc2015:  LIBS += -L../distrib -lcommon -lopengl32 -lGLU32
-win32-g++:  LIBS += -L../distrib -lcommon      
-linux:  LIBS += -L../distrib -lcommon
+win32-g++:  LIBS += -L../distrib -lcommon
+linux:  LIBS += -L../distrib -lcommon -lGLEW
 linux:QMAKE_RPATHDIR += ../distrib
 #CONFIG (release,debug | release) {
 #	win32-msvc2005:release:  LIBS += -L../common/release -lcommon
@@ -44,8 +44,6 @@
 win32-msvc2012:DEFINES += GLEW_STATIC
 win32-msvc2015:DEFINES += GLEW_STATIC
 
-#DEFINES += GLEW_STATIC
-
 # This removes the need of XSERVER to run meshlabserver
 DEFINES += NO_XSERVER_DEPENDENCY
 
