diff -Naur a/source/blender/editors/mesh/editmesh_add.c b/source/blender/editors/mesh/editmesh_add.c
--- a/source/blender/editors/mesh/editmesh_add.c	2020-12-24 20:07:32.000000000 +0300
+++ b/source/blender/editors/mesh/editmesh_add.c	2020-12-29 00:55:03.937332739 +0300
@@ -144,7 +144,7 @@
           "create_grid x_segments=%i y_segments=%i size=%f matrix=%m4 calc_uvs=%b",
           1,
           1,
-          RNA_float_get(op->ptr, "size") / 2.0f,
+          RNA_float_get(op->ptr, "size") / 0.2f,
           creation_data.mat,
           calc_uvs)) {
     return OPERATOR_CANCELLED;
@@ -387,7 +387,7 @@
   RNA_def_int(ot->srna, "vertices", 32, 3, MESH_ADD_VERTS_MAXI, "Vertices", "", 3, 500);
   ED_object_add_unit_props_radius(ot);
   RNA_def_float_distance(
-      ot->srna, "depth", 2.0f, 0.0, OBJECT_ADD_SIZE_MAXF, "Depth", "", 0.001, 100.00);
+      ot->srna, "depth", 0.2f, 0.0, OBJECT_ADD_SIZE_MAXF, "Depth", "", 0.001, 100.00);
   RNA_def_enum(ot->srna, "end_fill_type", fill_type_items, 1, "Cap Fill Type", "");
 
   ED_object_add_mesh_props(ot);
@@ -466,7 +466,7 @@
   RNA_def_float_distance(
       ot->srna, "radius2", 0.0f, 0.0, OBJECT_ADD_SIZE_MAXF, "Radius 2", "", 0.0, 100.00);
   RNA_def_float_distance(
-      ot->srna, "depth", 2.0f, 0.0, OBJECT_ADD_SIZE_MAXF, "Depth", "", 0.001, 100.00);
+      ot->srna, "depth", 0.2f, 0.0, OBJECT_ADD_SIZE_MAXF, "Depth", "", 0.001, 100.00);
   RNA_def_enum(ot->srna, "end_fill_type", fill_type_items, 1, "Base Fill Type", "");
 
   ED_object_add_mesh_props(ot);
@@ -507,7 +507,7 @@
           "create_grid x_segments=%i y_segments=%i size=%f matrix=%m4 calc_uvs=%b",
           RNA_int_get(op->ptr, "x_subdivisions"),
           RNA_int_get(op->ptr, "y_subdivisions"),
-          RNA_float_get(op->ptr, "size") / 2.0f,
+          RNA_float_get(op->ptr, "size") / 0.2f,
           creation_data.mat,
           calc_uvs)) {
     return OPERATOR_CANCELLED;
diff -Naur a/source/blender/editors/object/object_add.c b/source/blender/editors/object/object_add.c
--- a/source/blender/editors/object/object_add.c	2020-12-24 20:07:32.000000000 +0300
+++ b/source/blender/editors/object/object_add.c	2020-12-29 00:46:16.320656681 +0300
@@ -304,7 +304,7 @@
 void ED_object_add_unit_props_size(wmOperatorType *ot)
 {
   RNA_def_float_distance(
-      ot->srna, "size", 2.0f, 0.0, OBJECT_ADD_SIZE_MAXF, "Size", "", 0.001, 100.00);
+      ot->srna, "size", 0.2f, 0.0, OBJECT_ADD_SIZE_MAXF, "Size", "", 0.001, 100.00);
 }
 
 void ED_object_add_unit_props_radius_ex(wmOperatorType *ot, float default_value)
@@ -370,7 +370,7 @@
 
 void ED_object_add_mesh_props(wmOperatorType *ot)
 {
-  RNA_def_boolean(ot->srna, "calc_uvs", true, "Generate UVs", "Generate a default UV map");
+  RNA_def_boolean(ot->srna, "calc_uvs", false, "Generate UVs", "Generate a default UV map");
 }
 
 bool ED_object_add_generic_get_opts(bContext *C,
