--- a/src/apps/others/ojph_sockets.cpp
+++ b/src/apps/others/ojph_sockets.cpp
@@ -165,7 +165,7 @@ namespace ojph
                                   MAKELANGID(LANG_NEUTRAL, SUBLANG_DEFAULT), 
                                   buf, max_buf_size, NULL);
       buf[max_buf_size - 1] = 0;
-    #elif (defined OJPH_OS_APPLE) || \
+    #elif (defined OJPH_OS_LINUX) || \
       ((_POSIX_C_SOURCE >= 200112L) && !_GNU_SOURCE)
       // it is not clear if the returned value is in buf or in v
       int t = strerror_r(errnum, (char*)buf, max_buf_size);
