--- a/build_files/cmake/macros.cmake
+++ b/build_files/cmake/macros.cmake
@@ -547,7 +547,7 @@ macro(get_sse_flags
   if (CMAKE_SYSTEM_PROCESSOR MATCHES "(x86_64)|(AMD64)" OR CMAKE_OSX_ARCHITECTURES MATCHES x86_64)
     # message(STATUS "Detecting SSE support")
     if(CMAKE_COMPILER_IS_GNUCC OR (CMAKE_C_COMPILER_ID MATCHES "Clang"))
-      set(${_sse42_flags} "-march=x86-64-v2")
+      set(${_sse42_flags} "-msse4.2")
     elseif(MSVC)
       # MSVC has no specific compile flags for SSE42 (only for AVX).
       set(${_sse42_flags})
