--- a/build_files/cmake/platform/platform_unix.cmake
+++ b/build_files/cmake/platform/platform_unix.cmake
@@ -1022,13 +1022,6 @@ endif()
 
 unset(_IS_LINKER_DEFAULT)
 
-# Avoid conflicts with Mesa llvmpipe, Luxrender, and other plug-ins that may
-# use the same libraries as Blender with a different version or build options.
-set(PLATFORM_SYMBOLS_MAP ${CMAKE_SOURCE_DIR}/source/creator/symbols_unix.map)
-set(PLATFORM_LINKFLAGS
-  "${PLATFORM_LINKFLAGS} -Wl,--version-script='${PLATFORM_SYMBOLS_MAP}'"
-)
-
 # Don't use position independent executable for portable install since file
 # browsers can't properly detect blender as an executable then. Still enabled
 # for non-portable installs as typically used by Linux distributions.
