--- a/intern/CMakeLists.txt
+++ b/intern/CMakeLists.txt
@@ -32,7 +32,9 @@ if(WITH_BULLET)
 endif()
 
 if(WITH_DRACO)
-  add_subdirectory(draco_bridge)
+  if(NOT WITH_SYSTEM_DRACO)
+    add_subdirectory(draco_bridge)
+  endif()
 endif()
 
 if(WITH_MESHOPTIMIZER)
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -475,6 +475,10 @@ if(UNIX AND NOT APPLE)
 
 # Compression
 option(WITH_DRACO "Enable Draco mesh compression Python module (used for glTF)" ON)
+if(UNIX AND NOT APPLE)
+  option(WITH_SYSTEM_DRACO "Use the system DRACO library" OFF)
+  mark_as_advanced(WITH_SYSTEM_DRACO)
+endif()
 option(WITH_MESHOPTIMIZER "Enable meshoptimizer compression Python module (used for glTF)" ON)
 
 # Camera/motion tracking
--- a/source/creator/CMakeLists.txt
+++ b/source/creator/CMakeLists.txt
@@ -517,7 +517,7 @@ if(WITH_PYTHON)
     endif()
   endif()
 
-  if(WITH_DRACO)
+  if(WITH_DRACO AND NOT WITH_SYSTEM_DRACO)
     if(GLTF_BRIDGE_RPATH)
       set_target_properties(bf_intern_draco_bridge PROPERTIES
         BUILD_WITH_INSTALL_RPATH TRUE
