--- a/builds/cuda_nvcc/targets/x86_64-linux/include/crt/host_defines.h
+++ b/builds/cuda_nvcc/targets/x86_64-linux/include/crt/host_defines.h
@@ -79,8 +79,6 @@
    Hence, only define __noinline__ when the code is being processed
    by a  CUDA compiler component.
 */   
-#define __noinline__ \
-        __attribute__((noinline))
 #endif /* __CUDACC__  || __CUDA_ARCH__ || __CUDA_LIBDEVICE__ */
         
 #define __forceinline__ \
