diff -urp blender-3.5.1.orig/build_files/cmake/platform/platform_unix.cmake blender-3.5.1/build_files/cmake/platform/platform_unix.cmake
--- blender-3.5.1.orig/build_files/cmake/platform/platform_unix.cmake	2023-05-04 01:26:34.773826914 -0700
+++ blender-3.5.1/build_files/cmake/platform/platform_unix.cmake	2023-05-04 01:36:08.225841499 -0700
@@ -374,6 +374,12 @@ if(WITH_MATERIALX)
 endif()
 add_bundled_libraries(materialx/lib)
 
+find_package_wrapper(TBB2)
+if(NOT TBB2_FOUND)
+   message(WARNING "TBB2 not found, disabling WITH_USD")
+   set(WITH_USD OFF)
+endif()
+
 if(WITH_BOOST)
   # uses in build instructions to override include and library variables
   if(NOT BOOST_CUSTOM)
diff -urp blender-3.5.1.orig/source/blender/io/usd/CMakeLists.txt blender-3.5.1/source/blender/io/usd/CMakeLists.txt
--- blender-3.5.1.orig/source/blender/io/usd/CMakeLists.txt	2023-05-04 01:26:34.997826920 -0700
+++ blender-3.5.1/source/blender/io/usd/CMakeLists.txt	2023-05-04 01:31:51.073834959 -0700
@@ -55,7 +55,7 @@ set(INC
 set(INC_SYS
   ${USD_INCLUDE_DIRS}
   ${BOOST_INCLUDE_DIR}
-  ${TBB_INCLUDE_DIR}
+  ${TBB2_INCLUDE_DIR}
   ${PYTHON_INCLUDE_DIR}
   ${PYTHON_INCLUDE_DIRS}
 )
@@ -131,7 +131,7 @@ list(APPEND LIB
   ${BOOST_PYTHON_LIBRARIES}
   ${PYTHON_LIBRARIES}
   ${USD_LIBRARIES}
-  ${TBB_LIBRARIES}
+  ${TBB2_LIBRARIES}
   ${PYTHON_LINKFLAGS}
   ${PYTHON_LIBRARIES}
 )
