--- a/release/datafiles/locale
+++ b/release/datafiles/locale
@@ -1 +1 @@
-Subproject commit 2b3c19f5f61fc72dba56a7edfdc4e55e2327dc1f
--- a/release/scripts/addons
+++ b/release/scripts/addons
@@ -1 +1 @@
-Subproject commit 49c39f59fbc464dd34388990123f271c39eacbf4
--- a/release/scripts/addons_contrib
+++ b/release/scripts/addons_contrib
@@ -1 +1 @@
-Subproject commit a52733b58d95ce60ecde95a9eca242e7319c285a
--- a/source/tools
+++ b/source/tools
@@ -1 +1 @@
-Subproject commit 896c5f78952adb2d091d28c65086d46992dabdac
-- 
2.27.0

