--- a/build_files/cmake/platform/platform_unix.cmake.orig	2023-05-25 08:09:12.717335354 -0700
+++ b/build_files/cmake/platform/platform_unix.cmake	2023-05-25 08:10:09.753336516 -0700
@@ -963,4 +963,7 @@ function(CONFIGURE_ATOMIC_LIB_IF_NEEDED)
   endif()
 endfunction()
 
+list(APPEND CMAKE_INSTALL_RPATH ${TBB_LIB_DIR} ${TBB2_LIB_DIR})
+message(STATUS "TBB_LIB_DIR=${TBB_LIB_DIR} TBB2_LIB_DIR=${TBB2_LIB_DIR}")
+
 CONFIGURE_ATOMIC_LIB_IF_NEEDED()
