diff -urp blender-3.0.0.orig/CMakeLists.txt blender-3.0.0/CMakeLists.txt
--- blender-3.0.0.orig/CMakeLists.txt	2021-11-25 16:33:06.000000000 -0800
+++ blender-3.0.0/CMakeLists.txt	2021-12-05 18:07:54.092613009 -0800
@@ -133,14 +133,14 @@ enable_testing()
 #-----------------------------------------------------------------------------
 # Redirect output files
 
-set(EXECUTABLE_OUTPUT_PATH ${CMAKE_BINARY_DIR}/bin CACHE INTERNAL "" FORCE)
+set(EXECUTABLE_OUTPUT_PATH ${CMAKE_BINARY_DIR}/${CMAKE_INSTALL_BINDIR} CACHE INTERNAL "" FORCE)
 set(LIBRARY_OUTPUT_PATH ${CMAKE_BINARY_DIR}/lib CACHE INTERNAL "" FORCE)
 
 get_property(GENERATOR_IS_MULTI_CONFIG GLOBAL PROPERTY GENERATOR_IS_MULTI_CONFIG)
 if(GENERATOR_IS_MULTI_CONFIG)
-  set(TESTS_OUTPUT_DIR ${EXECUTABLE_OUTPUT_PATH}/tests/$<CONFIG>/ CACHE INTERNAL "" FORCE)
+  set(TESTS_OUTPUT_DIR ${CMAKE_BINARY_DIR}/tests/$<CONFIG>/ CACHE INTERNAL "" FORCE)
 else()
-  set(TESTS_OUTPUT_DIR ${EXECUTABLE_OUTPUT_PATH}/tests/ CACHE INTERNAL "" FORCE)
+  set(TESTS_OUTPUT_DIR ${CMAKE_BINARY_DIR}/tests/ CACHE INTERNAL "" FORCE)
 endif()
 
 #-----------------------------------------------------------------------------
Only in blender-3.0.0: CMakeLists.txt.orig
diff -urp blender-3.0.0.orig/source/creator/CMakeLists.txt blender-3.0.0/source/creator/CMakeLists.txt
--- blender-3.0.0.orig/source/creator/CMakeLists.txt	2021-12-05 18:07:15.917091556 -0800
+++ blender-3.0.0/source/creator/CMakeLists.txt	2021-12-05 18:08:38.169369904 -0800
@@ -531,7 +531,7 @@ if(UNIX AND NOT APPLE)
     # main blender binary
     install(
       TARGETS blender
-      DESTINATION bin
+      DESTINATION "${CMAKE_INSTALL_BINDIR}"
     )
     if(WITH_DOC_MANPAGE)
       # manpage only with 'blender' binary
@@ -557,7 +557,7 @@ if(UNIX AND NOT APPLE)
     if(WITH_BLENDER_THUMBNAILER)
       install(
         TARGETS blender-thumbnailer
-        DESTINATION bin
+        DESTINATION "${CMAKE_INSTALL_BINDIR}"
       )
     endif()
     set(BLENDER_TEXT_FILES_DESTINATION share/doc/blender)
