--- a/build_files/cmake/macros.cmake.orig	2022-05-02 06:58:34.000000000 -0700
+++ b/build_files/cmake/macros.cmake	2022-07-03 12:25:25.222953329 -0700
@@ -794,6 +794,7 @@ function(get_blender_version)
   # - BLENDER_VERSION_MINOR
   # - BLENDER_VERSION_PATCH
   # - BLENDER_VERSION_CYCLE (alpha, beta, rc, release)
+  # - BLENDER_VERSION_TAG (_filmic) to indicate type of mod
 
   # So cmake depends on BKE_blender.h, beware of inf-loops!
   CONFIGURE_FILE(${CMAKE_SOURCE_DIR}/source/blender/blenkernel/BKE_blender_version.h
@@ -804,6 +805,7 @@ function(get_blender_version)
   string(REGEX REPLACE ".*#define[ \t]+BLENDER_VERSION[ \t]+([0-9]+).*" "\\1" _out_version "${_contents}")
   string(REGEX REPLACE ".*#define[ \t]+BLENDER_VERSION_PATCH[ \t]+([0-9]+).*" "\\1" _out_version_patch "${_contents}")
   string(REGEX REPLACE ".*#define[ \t]+BLENDER_VERSION_CYCLE[ \t]+([a-z]+).*" "\\1" _out_version_cycle "${_contents}")
+  string(REGEX REPLACE ".*#define[ \t]+BLENDER_VERSION_TAG[ \t]+([a-z_]*).*" "\\1" _out_version_tag "${_contents}")
 
   if(NOT ${_out_version} MATCHES "[0-9]+")
     message(FATAL_ERROR "Version parsing failed for BLENDER_VERSION")
@@ -817,6 +819,10 @@ function(get_blender_version)
     message(FATAL_ERROR "Version parsing failed for BLENDER_VERSION_CYCLE")
   endif()
 
+  #if(NOT ${_out_version_tag} MATCHES "[a-z_]+")
+  #	message(FATAL_ERROR "Version parsing failed for BLENDER_VERSION_TAG")
+  #endif()
+
   math(EXPR _out_version_major "${_out_version} / 100")
   math(EXPR _out_version_minor "${_out_version} % 100")
 
@@ -826,6 +832,7 @@ function(get_blender_version)
   set(BLENDER_VERSION_MINOR "${_out_version_minor}" PARENT_SCOPE)
   set(BLENDER_VERSION_PATCH "${_out_version_patch}" PARENT_SCOPE)
   set(BLENDER_VERSION_CYCLE "${_out_version_cycle}" PARENT_SCOPE)
+  set(BLENDER_VERSION_TAG "${_out_version_tag}" PARENT_SCOPE)
 
 endfunction()
 
--- a/source/blender/blenkernel/BKE_blender_version.h.orig	2022-06-07 18:31:12.000000000 -0700
+++ b/source/blender/blenkernel/BKE_blender_version.h	2022-07-03 12:25:25.223953370 -0700
@@ -22,6 +22,8 @@ extern "C" {
 #define BLENDER_VERSION_PATCH 0
 /** Blender release cycle stage: alpha/beta/rc/release. */
 #define BLENDER_VERSION_CYCLE release
+/* tag or flavor */
+#define BLENDER_VERSION_TAG   
 
 /* Blender file format version. */
 #define BLENDER_FILE_VERSION BLENDER_VERSION
--- a/source/blender/blenkernel/intern/appdir.c.orig	2022-05-02 06:58:39.000000000 -0700
+++ b/source/blender/blenkernel/intern/appdir.c	2022-07-03 12:25:25.223953370 -0700
@@ -112,9 +112,9 @@ void BKE_appdir_exit(void)
  */
 static char *blender_version_decimal(const int version)
 {
-  static char version_str[5];
+  static char version_str[6];
   BLI_assert(version < 1000);
-  BLI_snprintf(version_str, sizeof(version_str), "%d.%d", version / 100, version % 100);
+  BLI_snprintf(version_str, sizeof(version_str), "%d.%d.%s%s", version / 100, version % 100, STRINGIFY(BLENDER_VERSION_PATCH), STRINGIFY(BLENDER_VERSION_TAG));
   return version_str;
 }
 
--- a/source/creator/CMakeLists.txt.orig	2022-07-03 12:23:54.518273428 -0700
+++ b/source/creator/CMakeLists.txt	2022-07-03 12:25:25.225953451 -0700
@@ -301,15 +301,15 @@ set(BLENDER_TEXT_FILES
 if(UNIX AND NOT APPLE)
   if(WITH_PYTHON_MODULE)
     if(WITH_INSTALL_PORTABLE)
-      set(TARGETDIR_VER ${BLENDER_VERSION})
+      set(TARGETDIR_VER ${BLENDER_VERSION}.${BLENDER_VERSION_PATCH}${BLENDER_VERSION_TAG})
     else()
-      set(TARGETDIR_VER ${PYTHON_SITE_PACKAGES}/${BLENDER_VERSION})
+      set(TARGETDIR_VER ${PYTHON_SITE_PACKAGES}/${BLENDER_VERSION}.${BLENDER_VERSION_PATCH}${BLENDER_VERSION_TAG})
     endif()
   else()
     if(WITH_INSTALL_PORTABLE)
-      set(TARGETDIR_VER ${BLENDER_VERSION})
+      set(TARGETDIR_VER ${BLENDER_VERSION}.${BLENDER_VERSION_PATCH}${BLENDER_VERSION_TAG})
     else()
-      set(TARGETDIR_VER share/blender/${BLENDER_VERSION})
+      set(TARGETDIR_VER share/blender/${BLENDER_VERSION}.${BLENDER_VERSION_PATCH}${BLENDER_VERSION_TAG})
     endif()
   endif()
 
@@ -813,7 +813,7 @@ elseif(WIN32)
 
       install(
         DIRECTORY ${LIBDIR}/python/${_PYTHON_VERSION_NO_DOTS}/lib
-        DESTINATION ${BLENDER_VERSION}/python/
+        DESTINATION ${BLENDER_VERSION}.${BLENDER_VERSION_PATCH}${BLENDER_VERSION_TAG}/python/
         CONFIGURATIONS Release;RelWithDebInfo;MinSizeRel
         PATTERN ".svn" EXCLUDE
         PATTERN "*_d.*" EXCLUDE                  # * debug libraries *
@@ -824,7 +824,7 @@ elseif(WIN32)
 
       install(
         DIRECTORY ${LIBDIR}/python/${_PYTHON_VERSION_NO_DOTS}/lib
-        DESTINATION ${BLENDER_VERSION}/python/
+        DESTINATION ${BLENDER_VERSION}.${BLENDER_VERSION_PATCH}${BLENDER_VERSION_TAG}/python/
         CONFIGURATIONS Debug
         PATTERN ".svn" EXCLUDE
         PATTERN "__pycache__" EXCLUDE           # * any cache *
@@ -834,7 +834,7 @@ elseif(WIN32)
 
       install(
         DIRECTORY ${LIBDIR}/python/${_PYTHON_VERSION_NO_DOTS}/DLLs
-        DESTINATION ${BLENDER_VERSION}/python
+        DESTINATION ${BLENDER_VERSION}.${BLENDER_VERSION_PATCH}${BLENDER_VERSION_TAG}/python
         CONFIGURATIONS Release;RelWithDebInfo;MinSizeRel
         PATTERN "*.pdb" EXCLUDE
         PATTERN "*_d.*" EXCLUDE
@@ -842,20 +842,20 @@ elseif(WIN32)
 
       install(
         DIRECTORY ${LIBDIR}/python/${_PYTHON_VERSION_NO_DOTS}/DLLs
-        DESTINATION ${BLENDER_VERSION}/python
+        DESTINATION ${BLENDER_VERSION}.${BLENDER_VERSION_PATCH}${BLENDER_VERSION_TAG}/python
         CONFIGURATIONS Debug
       )
 
       install(
         FILES ${LIBDIR}/python/${_PYTHON_VERSION_NO_DOTS}/bin/python${_PYTHON_VERSION_NO_DOTS}.dll
               ${LIBDIR}/python/${_PYTHON_VERSION_NO_DOTS}/bin/python.exe
-        DESTINATION ${BLENDER_VERSION}/python/bin
+        DESTINATION ${BLENDER_VERSION}.${BLENDER_VERSION_PATCH}${BLENDER_VERSION_TAG}/python/bin
         CONFIGURATIONS Release;RelWithDebInfo;MinSizeRel
       )
       install(
         FILES ${LIBDIR}/python/${_PYTHON_VERSION_NO_DOTS}/bin/python${_PYTHON_VERSION_NO_DOTS}_d.dll
               ${LIBDIR}/python/${_PYTHON_VERSION_NO_DOTS}/bin/python_d.exe
-        DESTINATION ${BLENDER_VERSION}/python/bin
+        DESTINATION ${BLENDER_VERSION}.${BLENDER_VERSION_PATCH}${BLENDER_VERSION_TAG}/python/bin
         CONFIGURATIONS Debug
       )
 
