--- a/intern/rigidbody/rb_bullet_api.cpp
+++ b/intern/rigidbody/rb_bullet_api.cpp
@@ -55,17 +55,17 @@
 
 #include "RBI_api.h"
 
-#include "btBulletDynamicsCommon.h"
+#include "bullet/btBulletDynamicsCommon.h"
 
-#include "LinearMath/btConvexHullComputer.h"
-#include "LinearMath/btMatrix3x3.h"
-#include "LinearMath/btScalar.h"
-#include "LinearMath/btTransform.h"
-#include "LinearMath/btVector3.h"
+#include "bullet/LinearMath/btConvexHullComputer.h"
+#include "bullet/LinearMath/btMatrix3x3.h"
+#include "bullet/LinearMath/btScalar.h"
+#include "bullet/LinearMath/btTransform.h"
+#include "bullet/LinearMath/btVector3.h"
 
-#include "BulletCollision/CollisionShapes/btScaledBvhTriangleMeshShape.h"
-#include "BulletCollision/Gimpact/btGImpactCollisionAlgorithm.h"
-#include "BulletCollision/Gimpact/btGImpactShape.h"
+#include "bullet/BulletCollision/CollisionShapes/btScaledBvhTriangleMeshShape.h"
+#include "bullet/BulletCollision/Gimpact/btGImpactCollisionAlgorithm.h"
+#include "bullet/BulletCollision/Gimpact/btGImpactShape.h"
 
 struct rbDynamicsWorld {
   btDiscreteDynamicsWorld *dynamicsWorld;
