diff -Naur a/source/blender/blenkernel/intern/blendfile.c b/source/blender/blenkernel/intern/blendfile.c
--- a/source/blender/blenkernel/intern/blendfile.c	2020-12-24 20:07:32.000000000 +0300
+++ b/source/blender/blenkernel/intern/blendfile.c	2020-12-29 01:23:49.910696788 +0300
@@ -600,7 +600,7 @@
   /* Add-ons. */
   {
     const char *addons[] = {
-        "io_anim_bvh",
+        "io_import_images_as_planes",
         "io_curve_svg",
         "io_mesh_ply",
         "io_mesh_stl",
@@ -610,6 +610,12 @@
         "io_scene_obj",
         "io_scene_x3d",
         "cycles",
+        "mesh_tools",
+        "mesh_f2",
+        "mesh_looptools",
+        "mesh_tiny_cad",
+        "node_arrange",
+        "node_wrangler",
     };
     for (int i = 0; i < ARRAY_SIZE(addons); i++) {
       bAddon *addon = BKE_addon_new();
