diff -Naur a/source/blender/makesdna/DNA_modifier_defaults.h b/source/blender/makesdna/DNA_modifier_defaults.h
--- a/source/blender/makesdna/DNA_modifier_defaults.h	2020-12-24 20:07:32.000000000 +0300
+++ b/source/blender/makesdna/DNA_modifier_defaults.h	2020-12-28 00:11:53.397550988 +0300
@@ -43,7 +43,7 @@
     .length = 0.0f, \
     .merge_dist = 0.01f, \
     .fit_type = MOD_ARR_FIXEDCOUNT, \
-    .offset_type = MOD_ARR_OFF_RELATIVE, \
+    .offset_type = MOD_ARR_OFF_CONST, \
     .flags = 0, \
     .count = 2, \
     .uv_offset = {0.0f, 0.0f}, \
@@ -51,8 +51,8 @@
 
 #define _DNA_DEFAULT_BevelModifierData \
   { \
-    .value = 0.1f, \
-    .res = 1, \
+    .value = 0.01f, \
+    .res = 2, \
     .flags = 0, \
     .val_flags = MOD_BEVEL_AMT_OFFSET, \
     .profile_type = MOD_BEVEL_PROFILE_SUPERELLIPSE, \
@@ -93,7 +93,7 @@
 #define _DNA_DEFAULT_CastModifierData \
   { \
     .object = NULL, \
-    .fac = 0.5f, \
+    .fac = 1.0f, \
     .radius = 0.0f, \
     .size = 0.0f, \
     .defgrp_name = "", \
@@ -259,7 +259,7 @@
 
 #define _DNA_DEFAULT_DecimateModifierData \
   { \
-    .percent = 1.0f, \
+    .percent = 0.1f, \
     .iter = 0, \
     .delimit = 0, \
     .symmetry_axis = 0, \
@@ -567,7 +567,7 @@
     .vgroup_name = "", \
     .factor = DEG2RADF(45.0f), \
     .limit = {0.0f, 1.0f}, \
-    .mode = MOD_SIMPLEDEFORM_MODE_TWIST, \
+    .mode = MOD_SIMPLEDEFORM_MODE_BEND, \
     .axis = 0, \
     .deform_axis = 0, \
     .flag = 0, \
@@ -608,7 +608,7 @@
     .crease_inner = 0.0f, \
     .crease_outer = 0.0f, \
     .crease_rim = 0.0f, \
-    .flag = MOD_SOLIDIFY_RIM, \
+    .flag = MOD_SOLIDIFY_RIM | MOD_SOLIDIFY_EVEN, \
     .mat_ofs = 0, \
     .mat_ofs_rim = 0, \
     .merge_tolerance = 0.0001f, \
diff -Naur a/source/blender/makesdna/DNA_object_defaults.h b/source/blender/makesdna/DNA_object_defaults.h
--- a/source/blender/makesdna/DNA_object_defaults.h	2020-12-24 20:07:32.000000000 +0300
+++ b/source/blender/makesdna/DNA_object_defaults.h	2020-12-27 22:56:45.667471288 +0300
@@ -56,7 +56,7 @@
     .dt = OB_TEXTURE, \
  \
     .empty_drawtype = OB_PLAINAXES, \
-    .empty_drawsize = 1.0, \
+    .empty_drawsize = 0.1, \
     .empty_image_depth = OB_EMPTY_IMAGE_DEPTH_DEFAULT, \
     .ima_ofs = {-0.5, -0.5}, \
  \
diff -Naur a/source/blender/makesdna/DNA_view3d_defaults.h b/source/blender/makesdna/DNA_view3d_defaults.h
--- a/source/blender/makesdna/DNA_view3d_defaults.h	2020-12-24 20:07:32.000000000 +0300
+++ b/source/blender/makesdna/DNA_view3d_defaults.h	2020-12-27 22:52:32.467466785 +0300
@@ -94,9 +94,9 @@
     .flag = V3D_SELECT_OUTLINE, \
     .flag2 = V3D_SHOW_RECONSTRUCTION | V3D_SHOW_ANNOTATION, \
  \
-    .lens = 50.0f, \
-    .clip_start = 0.01f, \
-    .clip_end = 1000.0f, \
+    .lens = 85.0f, \
+    .clip_start = 0.001f, \
+    .clip_end = 100.0f, \
  \
     .bundle_size = 0.2f, \
     .bundle_drawtype = OB_PLAINAXES, \
