diff --git a/intern/ghost/CMakeLists.txt b/intern/ghost/CMakeLists.txt
--- a/intern/ghost/CMakeLists.txt
+++ b/intern/ghost/CMakeLists.txt
@@ -245,10 +245,10 @@
     if(WITH_X11_XF86VMODE)
       add_definitions(-DWITH_X11_XF86VMODE)
       list(APPEND INC_SYS
-        ${X11_xf86vmode_INCLUDE_PATH}
+        ${X11_Xxf86vmode_INCLUDE_PATH}
       )
       list(APPEND LIB
-        ${X11_Xf86vmode_LIB}
+        ${X11_Xxf86vmode_LIB}
       )
     endif()


