--- a/source/blender/blenfont/intern/blf.c
+++ b/source/blender/blenfont/intern/blf.c
@@ -84,6 +84,7 @@
 static int global_font_default = -1;
 static int global_font_points = 11;
 static int global_font_dpi = 72;
+static bool global_use_hinting = true;
 
 /* XXX, should these be made into global_font_'s too? */
 int blf_mono_font = -1;
#diff -Naur ./orig/wm_files.c ./fix/wm_files.c
#--- a/source/blender/windowmanager/intern/wm_files.c
#+++ b/source/blender/windowmanager/intern/wm_files.c
#@@ -341,6 +341,7 @@
# 
# 	/* update tempdir from user preferences */
# 	BKE_tempdir_init(U.tempdir);
#+    BLF_hinting_set((U.text_render & USER_TEXT_DISABLE_HINTING) == 0);
# }
 
