--- a/release/scripts/addons/node_wrangler.py	2020-01-21 15:05:01.804392686 +0300
+++ b/release/scripts/addons/node_wrangler.py	2020-01-21 15:30:40.741064468 +0300
@@ -995,31 +995,31 @@
 class NWPrincipledPreferences(bpy.types.PropertyGroup):
     base_color: StringProperty(
         name='Base Color',
-        default='diffuse diff albedo base col color',
+        default='d diffuse diff albedo base col color',
         description='Naming Components for Base Color maps')
     sss_color: StringProperty(
         name='Subsurface Color',
-        default='sss subsurface',
+        default='sub sss subsurface',
         description='Naming Components for Subsurface Color maps')
     metallic: StringProperty(
         name='Metallic',
-        default='metallic metalness metal mtl',
+        default='m metallic metalness metal mtl',
         description='Naming Components for metallness maps')
     specular: StringProperty(
         name='Specular',
-        default='specularity specular spec spc',
+        default='s specularity specular spec spc',
         description='Naming Components for Specular maps')
     normal: StringProperty(
         name='Normal',
-        default='normal nor nrm nrml norm',
+        default='n normal nor nrm nrml norm',
         description='Naming Components for Normal maps')
     bump: StringProperty(
         name='Bump',
-        default='bump bmp',
+        default='b bump bmp',
         description='Naming Components for bump maps')
     rough: StringProperty(
         name='Roughness',
-        default='roughness rough rgh',
+        default='r roughness rough rgh',
         description='Naming Components for roughness maps')
     gloss: StringProperty(
         name='Gloss',
