reverted:
--- b/intern/cycles/integrator/path_trace.cpp
+++ a/intern/cycles/integrator/path_trace.cpp
@@ -355,9 +355,6 @@
 
   const int num_works = path_trace_works_.size();
 
-  tbb::task_group_context *tbb_ctx = tbb::task::self().group();
-  tbb_ctx->capture_fp_settings();
-
   tbb::parallel_for(0, num_works, [&](int i) {
     const double work_start_time = time_dt();
     const int num_samples = render_work.path_trace.num_samples;
