--- a/intern/cycles/integrator/path_trace.cpp	2022-04-14 19:41:26.832394102 +0300
+++ b/intern/cycles/integrator/path_trace.cpp	2022-04-14 19:45:08.689060985 +0300
@@ -355,8 +355,14 @@
 
   const int num_works = path_trace_works_.size();
 
-  tbb::task_group_context *tbb_ctx = tbb::task::self().group();
-  tbb_ctx->capture_fp_settings();
+  #if TBB_INTERFACE_VERSION_MAJOR >= 12
+  tbb::task_group_context *ctx = tbb::task::current_context();
+  if (ctx) {
+    ctx->cancel_group_execution();
+  }
+#else
+   tbb::task::self().cancel_group_execution();
+  #endif
 
   tbb::parallel_for(0, num_works, [&](int i) {
     const double work_start_time = time_dt();
