--- a/intern/ghost/CMakeLists.txt
+++ b/intern/ghost/CMakeLists.txt
@@ -183,6 +183,13 @@
       ${X11_X11_INCLUDE_PATH}
     )
 
+    list(APPEND LIB
+      ${X11_X11_LIB}
+      -lXext
+      -lXxf86vm
+      ${X11_Xrender_LIB}
+    )
+
     list(APPEND SRC
       intern/GHOST_DisplayManagerX11.cpp
       intern/GHOST_SystemX11.cpp
@@ -234,6 +241,9 @@
       list(APPEND INC_SYS
         ${X11_xf86vmode_INCLUDE_PATH}
       )
+      list(APPEND LIB
+        ${X11_Xf86vmode_LIB}
+      )
     endif()
 
     if(WITH_X11_XFIXES)
@@ -241,6 +251,9 @@
       list(APPEND INC_SYS
         ${X11_Xfixes_INCLUDE_PATH}
       )
+      list(APPEND LIB
+        ${X11_Xfixes_LIB}
+      )
     endif()
 
     if(WITH_X11_ALPHA)
@@ -252,6 +265,9 @@
       list(APPEND INC_SYS
         ${X11_Xinput_INCLUDE_PATH}
       )
+      list(APPEND LIB
+        ${X11_Xinput_LIB}
+      )
     endif()
 
     add_definitions(-DWITH_GHOST_X11)
