diff --git a/src/Gui/WidgetFactory.cpp b/src/Gui/WidgetFactory.cpp
index e9182de6d..7d67dd24d 100644
--- a/src/Gui/WidgetFactory.cpp
+++ b/src/Gui/WidgetFactory.cpp
@@ -77,7 +77,6 @@ PyTypeObject** SbkPySide_QtGuiTypes=NULL;
 # include <basewrapper.h>
 # include <sbkconverter.h>
 # include <sbkmodule.h>
-# include <typeresolver.h>
 # include <shiboken.h>
 # ifdef HAVE_PYSIDE2
 # define HAVE_PYSIDE
@@ -113,7 +112,7 @@ PyTypeObject** SbkPySide2_QtWidgetsTypes=NULL;
 
 using namespace Gui;
 
-#if defined (HAVE_SHIBOKEN)
+#if 0 // defined (HAVE_SHIBOKEN)
 namespace Shiboken {
 template<> struct Converter<Base::Quantity>
 {
@@ -206,7 +205,7 @@ void registerTypes()
 
 PythonWrapper::PythonWrapper()
 {
-#if defined (HAVE_SHIBOKEN)
+#if 0 // defined (HAVE_SHIBOKEN)
     static bool init = false;
     if (!init) {
         init = true;
@@ -296,7 +295,7 @@ QObject* PythonWrapper::toQObject(const Py::Object& pyobject)
 
 Py::Object PythonWrapper::fromQIcon(const QIcon* icon)
 {
-#if defined (HAVE_SHIBOKEN) && defined(HAVE_PYSIDE)
+#if 0 // defined (HAVE_SHIBOKEN) && defined(HAVE_PYSIDE)
     PyObject* pyobj = Shiboken::createWrapper<QIcon>(icon, true);
     if (pyobj)
         return Py::asObject(pyobj);
