--- a/node_wrangler.py	2021-03-04 02:25:21.923334214 +0300
+++ b/node_wrangler.py	2021-03-04 02:28:05.369326600 +0300
@@ -1111,7 +1111,7 @@
 class NWPrincipledPreferences(bpy.types.PropertyGroup):
     base_color: StringProperty(
         name='Base Color',
-        default='diffuse diff albedo base col color',
+        default='diffuse diff albedo base col color d D',
         description='Naming Components for Base Color maps')
     sss_color: StringProperty(
         name='Subsurface Color',
@@ -1119,7 +1119,7 @@
         description='Naming Components for Subsurface Color maps')
     metallic: StringProperty(
         name='Metallic',
-        default='metallic metalness metal mtl',
+        default='metallic metalness metal mtl m M',
         description='Naming Components for metallness maps')
     specular: StringProperty(
         name='Specular',
@@ -1127,7 +1127,7 @@
         description='Naming Components for Specular maps')
     normal: StringProperty(
         name='Normal',
-        default='normal nor nrm nrml norm',
+        default='normal nor nrm nrml norm n N',
         description='Naming Components for Normal maps')
     bump: StringProperty(
         name='Bump',
@@ -1135,7 +1135,7 @@
         description='Naming Components for bump maps')
     rough: StringProperty(
         name='Roughness',
-        default='roughness rough rgh',
+        default='roughness rough rgh r R',
         description='Naming Components for roughness maps')
     gloss: StringProperty(
         name='Gloss',
