From c75cfbdbc888a37319da4c6fa13cd21898b4eb1b Mon Sep 17 00:00:00 2001
From: Bernd Waibel <waebbl@gmail.com>
Date: Fri, 21 Feb 2020 17:44:34 +0100
Subject: [PATCH] CMakeLists.txt: Gentoo-specific patches

Signed-off-by: Bernd Waibel <waebbl@gmail.com>
---
 CMakeLists.txt | 24 +++++++++++++-----------
 1 file changed, 13 insertions(+), 11 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index ee1b532..52af1c3 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -43,7 +43,7 @@ else(APPLE)
   if(WIN32)
     set(INSTALL_DIR_DEFAULT "C:/netgen")
   else(WIN32)
-    set(INSTALL_DIR_DEFAULT /opt/netgen)
+    set(INSTALL_DIR_DEFAULT /usr)
   endif(WIN32)
 endif(APPLE)
 
@@ -52,14 +52,14 @@ if(INSTALL_DIR)
   set(INSTALL_DIR_DEFAULT ${INSTALL_DIR})
 endif(INSTALL_DIR)
 
-if(UNIX)
+if(UNIX AND NOT CMAKE_GENTOO_BUILD)
   message("Checking for write permissions in install directory...")
   execute_process(COMMAND mkdir -p ${CMAKE_INSTALL_PREFIX})
   execute_process(COMMAND test -w ${CMAKE_INSTALL_PREFIX} RESULT_VARIABLE res)
   if(res)
     message(WARNING "No write access at install directory, please set correct permissions")
   endif()
-endif(UNIX)
+endif(UNIX AND NOT CMAKE_GENTOO_BUILD)
 
 if (USE_SUPERBUILD)
   project (SUPERBUILD)
@@ -145,15 +145,15 @@ if(APPLE)
   set(NG_RPATH_TOKEN "@loader_path")
 else(APPLE)
   set(NG_INSTALL_DIR_BIN_DEFAULT bin)
-  set(NG_INSTALL_DIR_LIB_DEFAULT lib)
+  set(NG_INSTALL_DIR_LIB_DEFAULT lib${LIB_SUFFIX})
   if(WIN32)
     set(NG_INSTALL_DIR_CMAKE_DEFAULT cmake)
   else(WIN32)
-    set(NG_INSTALL_DIR_CMAKE_DEFAULT lib/cmake/${NG_INSTALL_SUFFIX})
+    set(NG_INSTALL_DIR_CMAKE_DEFAULT lib${LIB_SUFFIX}/cmake/${NG_INSTALL_SUFFIX})
   endif(WIN32)
   set(NG_INSTALL_DIR_PYTHON_DEFAULT ${PYTHON_PACKAGES_INSTALL_DIR})
   set(NG_INSTALL_DIR_RES_DEFAULT share)
-  set(NG_INSTALL_DIR_INCLUDE_DEFAULT include)
+  set(NG_INSTALL_DIR_INCLUDE_DEFAULT include/${NG_INSTALL_SUFFIX})
 
   set(NG_RPATH_TOKEN "\$ORIGIN")
 endif(APPLE)
@@ -338,10 +338,10 @@ endif (USE_MPEG)
 #######################################################################
 if(INSTALL_PROFILES)
     file(WRITE  ${CMAKE_CURRENT_BINARY_DIR}/netgen.sh "#!/bin/sh\n")
-    file(APPEND ${CMAKE_CURRENT_BINARY_DIR}/netgen.sh "export PATH=${CMAKE_INSTALL_PREFIX}/bin:$PATH\n")
+#    file(APPEND ${CMAKE_CURRENT_BINARY_DIR}/netgen.sh "export PATH=${CMAKE_INSTALL_PREFIX}/bin:$PATH\n")
     file(APPEND ${CMAKE_CURRENT_BINARY_DIR}/netgen.sh "export NETGENDIR=${CMAKE_INSTALL_PREFIX}/bin\n")
-    file(APPEND ${CMAKE_CURRENT_BINARY_DIR}/netgen.sh "export PYTHONPATH=${CMAKE_INSTALL_PREFIX}/${PYTHON_PACKAGES_INSTALL_DIR}:.\n")
-    file(APPEND ${CMAKE_CURRENT_BINARY_DIR}/netgen.sh "export LD_LIBRARY_PATH=${CMAKE_INSTALL_PREFIX}/lib:.\n")
+#    file(APPEND ${CMAKE_CURRENT_BINARY_DIR}/netgen.sh "export PYTHONPATH=${CMAKE_INSTALL_PREFIX}/${PYTHON_PACKAGES_INSTALL_DIR}:.\n")
+#    file(APPEND ${CMAKE_CURRENT_BINARY_DIR}/netgen.sh "export LD_LIBRARY_PATH=${CMAKE_INSTALL_PREFIX}/lib:.\n")
     install(PROGRAMS ${CMAKE_CURRENT_BINARY_DIR}/netgen.sh DESTINATION /etc/profile.d COMPONENT netgen)
 
     string(ASCII 27 Esc)
@@ -351,6 +351,7 @@ if(INSTALL_PROFILES)
 endif(INSTALL_PROFILES)
 
 #######################################################################
+if(APPLE)
 file(WRITE ${CMAKE_CURRENT_BINARY_DIR}/fixup.cmake "\
 include(${CMAKE_CURRENT_SOURCE_DIR}/cmake/netgen_fixup.cmake)
 set(APP ${CMAKE_INSTALL_PREFIX})
@@ -368,6 +369,7 @@ endif(NOT bundle_filename)
 execute_process(COMMAND hdiutil create -volname Netgen -srcfolder ${CMAKE_INSTALL_PREFIX} -ov -format UDZO \${bundle_filename}-${PACKAGE_VERSION}.dmg)
 ")
   add_custom_target(bundle COMMAND ${CMAKE_COMMAND} "-P" "${CMAKE_CURRENT_BINARY_DIR}/fixup.cmake")
+endif(APPLE)
 #######################################################################
 # CTest
 enable_testing()
@@ -415,7 +417,7 @@ add_subdirectory(tests)
 #######################################################################
 # Debian packager
 
-if(UNIX)
+if(UNIX AND NOT CMAKE_GENTOO_BUILD)
     set(CPACK_SOURCE_GENERATOR "TGZ")
     set(CPACK_SOURCE_IGNORE_FILES "/cmake/;/build/;/.gz/;~$;${CPACK_SOURCE_IGNORE_FILES}")
     set(CPACK_PACKAGE_VERSION ${PACKAGE_VERSION} )
@@ -446,7 +448,7 @@ if(UNIX)
         set(CPACK_DEBIAN_PACKAGE_NAME ${CPACK_PACKAGE_NAME})
         set(CPACK_PACKAGE_FILE_NAME "netgen-${PACKAGE_VERSION}_${CPACK_DEBIAN_PACKAGE_ARCHITECTURE}")   
     endif(temp)
-endif(UNIX)
+endif(UNIX AND NOT CMAKE_GENTOO_BUILD)
 
 if(APPLE)
     # create some auxiliary files
-- 
2.25.0

