--- a/intern/cycles/util/log.h	2023-02-05 23:32:04.863665393 +0300
+++ b/intern/cycles/util/log.h	2023-02-05 23:40:37.465247609 +0300
@@ -32,7 +32,7 @@ class LogMessageVoidify {
   }
 };
 
-#  define LOG_SUPPRESS() (true) ? ((void)0) : LogMessageVoidify() & StubStream()
+#  define LOG_SUPPRESS() (true) ? (void)0 : LogMessageVoidify() & StubStream()
 #  define LOG(severity) LOG_SUPPRESS()
 #  define VLOG(severity) LOG_SUPPRESS()
 #  define VLOG_IF(severity, condition) LOG_SUPPRESS()
