--- include/asm-generic/qrwlock.h
+++ include/asm-generic/qrwlock.h
@@ -89,7 +89,7 @@
 	if (unlikely(cnts))
 		return 0;
 
-	return likely(atomic_cmpxchg_acquire(&lock->cnts,
+	return likely((u32)atomic_cmpxchg_acquire(&lock->cnts,
 					     cnts, cnts | _QW_LOCKED) == cnts);
 }
 /**
--- arch/x86/include/asm/uaccess.h
+++ arch/x86/include/asm/uaccess.h
@@ -709,7 +709,7 @@
 	 * case, and do only runtime checking for non-constant sizes.
 	 */
 
-	if (likely(sz < 0 || sz >= n))
+	if (likely(sz < 0 || (unsigned long)sz >= n))
 		n = _copy_from_user(to, from, n);
 	else if(__builtin_constant_p(n))
 		copy_from_user_overflow();
@@ -727,7 +727,7 @@
 	might_fault();
 
 	/* See the comment in copy_from_user() above. */
-	if (likely(sz < 0 || sz >= n))
+	if (likely(sz < 0 || (unsigned long)sz >= n))
 		n = _copy_to_user(to, from, n);
 	else if(__builtin_constant_p(n))
 		copy_to_user_overflow();
