--- a/src/liboslexec/llvm_util.cpp	2022-03-29 01:46:49.000000000 -0000
+++ b/src/liboslexec/llvm_util.cpp	2022-04-11 09:36:04.491161194 -0000
@@ -45,7 +45,7 @@
 #include <llvm/Support/raw_os_ostream.h>
 #include <llvm/IR/LegacyPassManager.h>
 #include <llvm/IR/ValueSymbolTable.h>
-#include <llvm/Support/TargetRegistry.h>
+#include <llvm/MC/TargetRegistry.h>
 
 #include <llvm/Bitcode/BitcodeReader.h>
 #include <llvm/Bitcode/BitcodeWriter.h>
@@ -1677,7 +1677,7 @@ LLVM_Util::setup_optimization_passes (in
         builder.DisableUnrollLoops = false;
         builder.SLPVectorize = false;
         builder.LoopVectorize = false;
-        builder.DisableTailCalls = false;
+        // builder.DisableTailCalls = false;
 
         if (target_machine)
             target_machine->adjustPassManager(builder);
