--- CMakeLists.txt	2015-10-16 17:26:52.000000000 +0300
+++ CMakeLists.txt	2015-11-06 20:28:21.488604569 +0300
@@ -107,11 +107,7 @@
 # Bidirectional support library
 add_subdirectory(src/libbidir)
 # Python binding library
-if (BUILD_PYTHON)
-  add_subdirectory(src/libpython)
-elseif(NOT PYTHON_FOUND)
-  message(STATUS "Python was not found. The bindings will not be built.")
-endif()
+add_subdirectory(src/libpython)
 
 
 # Additional files to add to main executables
