diff --git a/imagepaste/clipboard/linux/linux.py b/imagepaste/clipboard/linux/linux.py
index aa67bd5..7238ad0 100644
--- a/imagepaste/clipboard/linux/linux.py
+++ b/imagepaste/clipboard/linux/linux.py
@@ -117,9 +117,8 @@ class LinuxClipboard(Clipboard):
         from os.path import realpath
         from stat import S_IXUSR
 
-        XCLIP_PATH = f"{dirname(realpath(__file__))}/bin/xclip"
+        XCLIP_PATH = f"/usr/bin/xclip"
         # TODO: Move to higher level
-        chmod(XCLIP_PATH, S_IXUSR)
         args = [
             XCLIP_PATH,
             "-rmlastnl",
