diff -Nru a/openvdb/python/pyFloatGrid.cc b/openvdb/python/pyFloatGrid.cc
--- a/openvdb/python/pyFloatGrid.cc	2018-04-10 21:22:17.000000000 +0300
+++ b/openvdb/python/pyFloatGrid.cc	2018-04-14 06:57:20.954716135 +0300
@@ -45,7 +45,7 @@
     // Add a module-level list that gives the types of all supported Grid classes.
     py::scope().attr("GridTypes") = py::list();
 
-#if defined(PY_OPENVDB_USE_NUMPY) && !defined(PY_OPENVDB_USE_BOOST_PYTHON_NUMPY)
+#if BOOST_VERSION < 106500 && defined(PY_OPENVDB_USE_NUMPY) && !defined(PY_OPENVDB_USE_BOOST_PYTHON_NUMPY)
     // Specify that py::numeric::array should refer to the Python type numpy.ndarray
     // (rather than the older Numeric.array).
     py::numeric::array::set_module_and_type("numpy", "ndarray");
diff -Nru a/openvdb/python/test/TestOpenVDB.py b/openvdb/python/test/TestOpenVDB.py
--- a/openvdb/python/test/TestOpenVDB.py	2018-04-10 21:22:17.000000000 +0300
+++ b/openvdb/python/test/TestOpenVDB.py	2018-04-14 06:56:11.578046769 +0300
@@ -533,7 +533,7 @@
                 for dtype in (np.float32, np.int32, np.float64, np.int64, np.uint32, np.bool):
                     # Create a NumPy array, fill it with the background value,
                     # then set some elements to the foreground value.
-                    arr = np.ndarray(shape, dtype)
+                    arr = np.zeros(shape, dtype)
                     arr.fill(BG)
                     bg = arr[0, 0, 0]
                     for c in coords:
@@ -628,7 +628,7 @@
             ):
                 for dtype in (np.float32, np.int32, np.float64, np.int64, np.uint32, np.bool):
                     # Return a new NumPy array.
-                    arr = np.ndarray(shape, dtype)
+                    arr = np.zeros(shape, dtype)
                     arr.fill(-100)
                     yield arr
 
