From 7bbca8c1e2babc941fea58f6aea203ba855a6aab Mon Sep 17 00:00:00 2001
From: pon <49616225+AJpon@users.noreply.github.com>
Date: Sat, 6 Apr 2024 15:31:59 +0900
Subject: [PATCH] Fix /openvdb/nanovdb/nanovdb/CMakeLists.txt

Signed-off-by: pon <49616225+AJpon@users.noreply.github.com>
---
 nanovdb/nanovdb/CMakeLists.txt | 26 +++++++++++++++-----------
 1 file changed, 15 insertions(+), 11 deletions(-)

diff --git a/nanovdb/nanovdb/CMakeLists.txt b/nanovdb/nanovdb/CMakeLists.txt
index 7bb3ab862d..406776cdaa 100644
--- a/nanovdb/nanovdb/CMakeLists.txt
+++ b/nanovdb/nanovdb/CMakeLists.txt
@@ -169,17 +169,6 @@ set(NANOVDB_INCLUDE_UTILFILES
   util/CpuTimer.h
   util/CreateNanoGrid.h
   util/CSampleFromVoxels.h
-  util/cuda/CudaAddBlindData.cuh
-  util/cuda/CudaDeviceBuffer.h
-  util/cuda/CudaGridChecksum.cuh
-  util/cuda/CudaGridHandle.cuh
-  util/cuda/CudaGridStats.cuh
-  util/cuda/CudaIndexToGrid.cuh
-  util/cuda/CudaNodeManager.cuh
-  util/cuda/CudaPointsToGrid.cuh
-  util/cuda/CudaSignedFloodFill.cuh
-  util/cuda/CudaUtils.h
-  util/cuda/GpuTimer.h
   util/DitherLUT.h
   util/ForEach.h
   util/GridBuilder.h
@@ -202,6 +191,19 @@ set(NANOVDB_INCLUDE_UTILFILES
   util/SampleFromVoxels.h
   util/Stencils.h
 )
+set(NANOVDB_INCLUDE_UTILFILES_CUDA
+  util/cuda/CudaAddBlindData.cuh
+  util/cuda/CudaDeviceBuffer.h
+  util/cuda/CudaGridChecksum.cuh
+  util/cuda/CudaGridHandle.cuh
+  util/cuda/CudaGridStats.cuh
+  util/cuda/CudaIndexToGrid.cuh
+  util/cuda/CudaNodeManager.cuh
+  util/cuda/CudaPointsToGrid.cuh
+  util/cuda/CudaSignedFloodFill.cuh
+  util/cuda/CudaUtils.h
+  util/cuda/GpuTimer.h
+)
 
 add_library(nanovdb INTERFACE)
 target_include_directories(nanovdb INTERFACE ../)
@@ -268,9 +270,11 @@ endif()
 
 set(NANOVDB_INSTALL_INCLUDEDIR ${CMAKE_INSTALL_INCLUDEDIR}/nanovdb)
 set(NANOVDB_INSTALL_UTILDIR ${NANOVDB_INSTALL_INCLUDEDIR}/util)
+set(NANOVDB_INSTALL_UTILDIR_CUDA ${NANOVDB_INSTALL_UTILDIR}/cuda)
 
 install(FILES ${NANOVDB_INCLUDE_FILES} DESTINATION ${NANOVDB_INSTALL_INCLUDEDIR})
 install(FILES ${NANOVDB_INCLUDE_UTILFILES} DESTINATION ${NANOVDB_INSTALL_UTILDIR})
+install(FILES ${NANOVDB_INCLUDE_UTILFILES_CUDA} DESTINATION ${NANOVDB_INSTALL_UTILDIR_CUDA})
 
 ###############################################################################
 # Options
