--- a/include/host_config.h
+++ b/include/host_config.h
@@ -114,11 +114,11 @@
 
 #if defined(__GNUC__)
 
-#if __GNUC__ > 5
+#if __GNUC__ > 7
 
-#error -- unsupported GNU version! gcc versions later than 5 are not supported!
+#error -- unsupported GNU version! gcc versions later than 8 are not supported!
 
-#endif /* __GNUC__ > 5 */
+#endif /* __GNUC__ > 7 */
 
 #if defined(__APPLE__) && defined(__MACH__) && !defined(__clang__)
 #error -- clang and clang++ are the only supported host compilers on Mac OS X!
