--- qt/python.pri	2015-09-07 12:57:03.000000000 +0300
+++ qt/python.pri	2015-09-07 16:00:05.542693570 +0300
@@ -1,25 +1,25 @@
 # Platform-specific configuration common to Antimony and supporting libraries
 cygwin {
-    QMAKE_CXXFLAGS += $$system(python3-config --includes)
-    QMAKE_CFLAGS += $$system(python3-config --includes)
-    LIBS += $$system(python3-config --libs)
+    QMAKE_CXXFLAGS += $$system(python-config-3.4 --includes)
+    QMAKE_CFLAGS += $$system(python-config-3.4 --includes)
+    LIBS += $$system(python-config-3.4 --libs)
     LIBS += -lboost_python3
 }
 
 macx {
-    QMAKE_CXXFLAGS += $$system(/usr/local/bin/python3-config --includes)
-    QMAKE_LFLAGS   += $$system(/usr/local/bin/python3-config --ldflags)
+    QMAKE_CXXFLAGS += $$system(/usr/local/bin/python-config-3.4 --includes)
+    QMAKE_LFLAGS   += $$system(/usr/local/bin/python-config-3.4 --ldflags)
     LIBS += -L/usr/local/lib -lboost_python3
     QMAKE_CXXFLAGS += -isystem/usr/local/include
 }
 
 linux {
-    QMAKE_CXXFLAGS += $$system(/usr/bin/python3-config --includes)
-    QMAKE_LFLAGS   += $$system(/usr/bin/python3-config --ldflags)
+    QMAKE_CXXFLAGS += $$system(/usr/bin/python-config-3.4 --includes)
+    QMAKE_LFLAGS   += $$system(/usr/bin/python-config-3.4 --ldflags)
 
     # Even though this is in QMAKE_LFLAGS, the linker is picky about
     # library ordering (so it needs to be here too).
-    LIBS += -lpython3.4m
+    LIBS += -lpython3.4
 
     # ldconfig is being used to find libboost_python, but it's in a different
     # place in different distros (and is not in the default $PATH on Debian).
@@ -40,7 +40,7 @@
 
     # Check for different boost::python naming schemes
     LDCONFIG_OUT = $$system($$LDCONFIG_BIN -p|grep python)
-    for (b, $$list(boost_python-py34 boost_python3)) {
+    for (b, $$list(boost_python-py34 boost_python-3.4)) {
         contains(LDCONFIG_OUT, "lib$${b}.so") {
             LIBS += "-l$$b"
             GOT_BOOST_PYTHON = True
