diff --git a/CMakeLists.txt b/CMakeLists.txt
index a325f281..2d8c5665 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -16,20 +16,13 @@ endif()
 
 if (WIN32)
     find_package(Boost REQUIRED COMPONENTS python3)
-elseif (APPLE)
+else ()
     foreach (PYTHON_NAME python3 python36 python37)
         find_package(Boost QUIET COMPONENTS ${PYTHON_NAME})
         if (${Boost_FOUND})
             break()
         endif()
     endforeach()
-elseif (UNIX)
-    foreach (PYTHON_NAME python3 python-py36 python-py35 python-py34)
-        find_package(Boost QUIET COMPONENTS ${PYTHON_NAME})
-        if (${Boost_FOUND})
-            break()
-        endif()
-    endforeach()
 endif()
 
 if (NOT ${Boost_FOUND})
diff --git a/lib/fab/src/tree/v2syntax.y b/lib/fab/src/tree/v2syntax.y
index d5d4f3b3..61f37fb1 100644
--- a/lib/fab/src/tree/v2syntax.y
+++ b/lib/fab/src/tree/v2syntax.y
@@ -37,15 +37,6 @@
 %syntax_error
 {
 	environment->valid = false;
-#ifndef NDEBUG
-	int n = sizeof(yyTokenName) / sizeof(yyTokenName[0]);
-	for (int i = 0; i < n; ++i) {
-		int a = yy_find_shift_action(yypParser, (YYCODETYPE)i);
-		if (a < YYNSTATE + YYNRULE) {
-		printf("possible token: %s\n", yyTokenName[i]);
-		}
-	}
-#endif
 }
