Index: source/blender/editors/transform/transform_ops.c
===================================================================
--- a/source/blender/editors/transform/transform_ops.c
+++ b/source/blender/editors/transform/transform_ops.c
@@ -161,11 +161,14 @@
 static int select_orientation_exec(bContext *C, wmOperator *op)
 {
 	Scene *scene = CTX_data_scene(C);
+	View3D *v3d = CTX_wm_view3d(C);
+
 	int orientation = RNA_enum_get(op->ptr, "orientation");
 
 	BKE_scene_orientation_slot_set_index(&scene->orientation_slots[SCE_ORIENT_DEFAULT], orientation);
 
 	WM_event_add_notifier(C, NC_SCENE | ND_TOOLSETTINGS, NULL);
+	WM_event_add_notifier(C, NC_SPACE | ND_SPACE_VIEW3D, v3d);
 
 	struct wmMsgBus *mbus = CTX_wm_message_bus(C);
 	WM_msg_publish_rna_prop(mbus, &scene->id, scene, TransformOrientationSlot, type);
