diff --git a/PIL/Image.py b/PIL/Image.py
index e2540a2..983ed5a 100644
--- a/PIL/Image.py
+++ b/PIL/Image.py
@@ -91,7 +91,7 @@ try:
     # import Image and use the Image.core variable instead.
     # Also note that Image.core is not a publicly documented interface,
     # and should be considered private and subject to change.
-    from . import _imaging as core
+    from .Image import core
 
     if __version__ != getattr(core, "PILLOW_VERSION", None):
         raise ImportError(
