diff -urp blender-3.6.0.orig/build_files/cmake/macros.cmake blender-3.6.0/build_files/cmake/macros.cmake
--- blender-3.6.0.orig/build_files/cmake/macros.cmake	2023-05-17 16:38:28.000000000 -0700
+++ blender-3.6.0/build_files/cmake/macros.cmake	2023-07-12 20:59:51.346230898 -0700
@@ -822,6 +822,7 @@ function(get_blender_version)
   # - BLENDER_VERSION_MINOR
   # - BLENDER_VERSION_PATCH
   # - BLENDER_VERSION_CYCLE (alpha, beta, rc, release)
+  # - BLENDER_VERSION_TAG (_filmic) to indicate type of mod
 
   # So CMAKE depends on `BKE_blender.h`, beware of infinite-loops!
   configure_file(
@@ -834,6 +835,7 @@ function(get_blender_version)
   string(REGEX REPLACE ".*#define[ \t]+BLENDER_VERSION[ \t]+([0-9]+).*" "\\1" _out_version "${_contents}")
   string(REGEX REPLACE ".*#define[ \t]+BLENDER_VERSION_PATCH[ \t]+([0-9]+).*" "\\1" _out_version_patch "${_contents}")
   string(REGEX REPLACE ".*#define[ \t]+BLENDER_VERSION_CYCLE[ \t]+([a-z]+).*" "\\1" _out_version_cycle "${_contents}")
+  string(REGEX REPLACE ".*#define[ \t]+BLENDER_VERSION_TAG[ \t]+([a-z_]*).*" "\\1" _out_version_tag "${_contents}")
 
   if(NOT ${_out_version} MATCHES "[0-9]+")
     message(FATAL_ERROR "Version parsing failed for BLENDER_VERSION")
@@ -847,6 +849,10 @@ function(get_blender_version)
     message(FATAL_ERROR "Version parsing failed for BLENDER_VERSION_CYCLE")
   endif()
 
+  #if(NOT ${_out_version_tag} MATCHES "[a-z_]+")
+  #	message(FATAL_ERROR "Version parsing failed for BLENDER_VERSION_TAG")
+  #endif()
+
   math(EXPR _out_version_major "${_out_version} / 100")
   math(EXPR _out_version_minor "${_out_version} % 100")
 
@@ -856,6 +862,7 @@ function(get_blender_version)
   set(BLENDER_VERSION_MINOR "${_out_version_minor}" PARENT_SCOPE)
   set(BLENDER_VERSION_PATCH "${_out_version_patch}" PARENT_SCOPE)
   set(BLENDER_VERSION_CYCLE "${_out_version_cycle}" PARENT_SCOPE)
+  set(BLENDER_VERSION_TAG "${_out_version_tag}" PARENT_SCOPE)
 
 endfunction()
 
Only in blender-3.6.0/build_files/cmake: macros.cmake.orig
diff -urp blender-3.6.0.orig/source/blender/blenkernel/BKE_blender_version.h blender-3.6.0/source/blender/blenkernel/BKE_blender_version.h
--- blender-3.6.0.orig/source/blender/blenkernel/BKE_blender_version.h	2023-06-26 16:30:35.000000000 -0700
+++ blender-3.6.0/source/blender/blenkernel/BKE_blender_version.h	2023-07-12 20:59:51.346230898 -0700
@@ -22,6 +22,8 @@ extern "C" {
 #define BLENDER_VERSION_PATCH 0
 /** Blender release cycle stage: alpha/beta/rc/release. */
 #define BLENDER_VERSION_CYCLE release
+/* tag or flavor */
+#define BLENDER_VERSION_TAG   
 
 /* Blender file format version. */
 #define BLENDER_FILE_VERSION BLENDER_VERSION
Only in blender-3.6.0/source/blender/blenkernel: BKE_blender_version.h.orig
diff -urp blender-3.6.0.orig/source/blender/blenkernel/intern/appdir.c blender-3.6.0/source/blender/blenkernel/intern/appdir.c
--- blender-3.6.0.orig/source/blender/blenkernel/intern/appdir.c	2023-05-17 16:38:29.000000000 -0700
+++ blender-3.6.0/source/blender/blenkernel/intern/appdir.c	2023-07-12 21:00:58.382229164 -0700
@@ -113,9 +113,9 @@ void BKE_appdir_exit(void)
  */
 static char *blender_version_decimal(const int version)
 {
-  static char version_str[5];
+  static char version_str[6];
   BLI_assert(version < 1000);
-  SNPRINTF(version_str, "%d.%d", version / 100, version % 100);
+  SNPRINTF(version_str, "%d.%d.%s%s", version / 100, version % 100, STRINGIFY(BLENDER_VERSION_PATCH), STRINGIFY(BLENDER_VERSION_TAG));
   return version_str;
 }
 
Only in blender-3.6.0/source/blender/blenkernel/intern: appdir.c.orig
Only in blender-3.6.0/source/blender/blenkernel/intern: appdir.c.rej
diff -urp blender-3.6.0.orig/source/creator/CMakeLists.txt blender-3.6.0/source/creator/CMakeLists.txt
--- blender-3.6.0.orig/source/creator/CMakeLists.txt	2023-07-12 20:59:32.310231390 -0700
+++ blender-3.6.0/source/creator/CMakeLists.txt	2023-07-12 20:59:51.346230898 -0700
@@ -353,20 +353,20 @@ if(UNIX AND NOT APPLE)
   if(WITH_PYTHON_MODULE)
     if(WITH_INSTALL_PORTABLE)
       set(TARGETDIR_BPY bpy)
-      set(TARGETDIR_VER bpy/${BLENDER_VERSION})
+      set(TARGETDIR_VER bpy/${BLENDER_VERSION}.${BLENDER_VERSION_PATCH}${BLENDER_VERSION_TAG})
       set(TARGETDIR_LIB bpy/lib)
     else()
       set(TARGETDIR_BPY ${PYTHON_SITE_PACKAGES}/bpy)
-      set(TARGETDIR_VER ${PYTHON_SITE_PACKAGES}/bpy/${BLENDER_VERSION})
+      set(TARGETDIR_VER ${PYTHON_SITE_PACKAGES}/bpy/${BLENDER_VERSION}.${BLENDER_VERSION_PATCH}${BLENDER_VERSION_TAG})
       set(TARGETDIR_LIB ${PYTHON_SITE_PACKAGES}/bpy/lib)
     endif()
   else()
     if(WITH_INSTALL_PORTABLE)
-      set(TARGETDIR_VER ${BLENDER_VERSION})
+      set(TARGETDIR_VER ${BLENDER_VERSION}.${BLENDER_VERSION_PATCH}${BLENDER_VERSION_TAG})
       set(TARGETDIR_TEXT .)
       set(TARGETDIR_LIB lib)
     else()
-      set(TARGETDIR_VER share/blender/${BLENDER_VERSION})
+      set(TARGETDIR_VER share/blender/${BLENDER_VERSION}.${BLENDER_VERSION_PATCH}${BLENDER_VERSION_TAG})
       set(TARGETDIR_TEXT share/doc/blender)
     endif()
   endif()
@@ -1074,7 +1074,7 @@ elseif(WIN32)
 
       install(
         DIRECTORY ${LIBDIR}/python/${_PYTHON_VERSION_NO_DOTS}/lib
-        DESTINATION ${BLENDER_VERSION}/python/
+        DESTINATION ${BLENDER_VERSION}.${BLENDER_VERSION_PATCH}${BLENDER_VERSION_TAG}/python/
         CONFIGURATIONS Release;RelWithDebInfo;MinSizeRel
         PATTERN ".svn" EXCLUDE
         PATTERN "*_d.*" EXCLUDE                 # * debug libraries *
@@ -1085,7 +1085,7 @@ elseif(WIN32)
 
       install(
         DIRECTORY ${LIBDIR}/python/${_PYTHON_VERSION_NO_DOTS}/lib
-        DESTINATION ${BLENDER_VERSION}/python/
+        DESTINATION ${BLENDER_VERSION}.${BLENDER_VERSION_PATCH}${BLENDER_VERSION_TAG}/python/
         CONFIGURATIONS Debug
         PATTERN ".svn" EXCLUDE
         PATTERN "__pycache__" EXCLUDE           # * any cache *
@@ -1095,7 +1095,7 @@ elseif(WIN32)
 
       install(
         DIRECTORY ${LIBDIR}/python/${_PYTHON_VERSION_NO_DOTS}/DLLs
-        DESTINATION ${BLENDER_VERSION}/python
+        DESTINATION ${BLENDER_VERSION}.${BLENDER_VERSION_PATCH}${BLENDER_VERSION_TAG}/python
         CONFIGURATIONS Release;RelWithDebInfo;MinSizeRel
         PATTERN "*.pdb" EXCLUDE
         PATTERN "*_d.*" EXCLUDE
@@ -1103,7 +1103,7 @@ elseif(WIN32)
 
       install(
         DIRECTORY ${LIBDIR}/python/${_PYTHON_VERSION_NO_DOTS}/DLLs
-        DESTINATION ${BLENDER_VERSION}/python
+        DESTINATION ${BLENDER_VERSION}.${BLENDER_VERSION_PATCH}${BLENDER_VERSION_TAG}/python
         CONFIGURATIONS Debug
       )
 
@@ -1111,14 +1111,14 @@ elseif(WIN32)
         FILES
           ${LIBDIR}/python/${_PYTHON_VERSION_NO_DOTS}/bin/python${_PYTHON_VERSION_NO_DOTS}.dll
           ${LIBDIR}/python/${_PYTHON_VERSION_NO_DOTS}/bin/python.exe
-        DESTINATION ${BLENDER_VERSION}/python/bin
+        DESTINATION ${BLENDER_VERSION}.${BLENDER_VERSION_PATCH}${BLENDER_VERSION_TAG}/python/bin
         CONFIGURATIONS Release;RelWithDebInfo;MinSizeRel
       )
       install(
         FILES
           ${LIBDIR}/python/${_PYTHON_VERSION_NO_DOTS}/bin/python${_PYTHON_VERSION_NO_DOTS}_d.dll
           ${LIBDIR}/python/${_PYTHON_VERSION_NO_DOTS}/bin/python_d.exe
-        DESTINATION ${BLENDER_VERSION}/python/bin
+        DESTINATION ${BLENDER_VERSION}.${BLENDER_VERSION_PATCH}${BLENDER_VERSION_TAG}/python/bin
         CONFIGURATIONS Debug
       )
 
Only in blender-3.6.0/source/creator: CMakeLists.txt.orig
