From a793c7d67fc886b1eb66ab67551f3a2448e4adc0 Mon Sep 17 00:00:00 2001
From: Matthew Smith <matthew@gentoo.org>
Date: Thu, 9 Jun 2022 20:19:19 +0100
Subject: [PATCH] build: Use system mimalloc

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -185,25 +185,7 @@ endif()
     add_subdirectory(extlib/libdxfrw)
 endif()
 
-message(STATUS "Using in-tree mimalloc")
-set(MI_OVERRIDE OFF CACHE BOOL "")
-set(MI_BUILD_SHARED OFF CACHE BOOL "")
-set(MI_BUILD_OBJECT OFF CACHE BOOL "")
-set(MI_BUILD_TESTS OFF CACHE BOOL "")
-if(EMSCRIPTEN)
-    # HACK: We only use mimalloc for the temporary heap, not as a general purpose
-    # allocator. However, the Emscripten port of mimalloc assumes that it is used
-    # as the system allocator, and depends on the slow emmalloc allocator.
-    # We do not want to enable emmalloc, as that would slow down allocations all
-    # over the place, so we must either use mimalloc as the system allocator as
-    # well, or override its definitions to just use the existing system allocator.
-    # The latter option is the least intrusive change from a stability standpoint,
-    # so for now that's what we do here. We can try to convert to mimalloc wholsale
-    # and remove this hack if we find the need to do so in the future.
-    set(MI_EXTRA_CPPDEFS "emmalloc_free=free;emmalloc_memalign=memalign" CACHE INTERNAL "")
-endif()
-add_subdirectory(extlib/mimalloc EXCLUDE_FROM_ALL)
-set(MIMALLOC_INCLUDE_DIR ${CMAKE_SOURCE_DIR}/extlib/mimalloc/include)
+find_package(mimalloc REQUIRED)
 
 if(NOT FORCE_VENDORED_Eigen3)
     find_package(Eigen3 CONFIG)
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -27,7 +27,6 @@ target_include_directories(slvs_deps INTERFACE SYSTEM
     ${ZLIB_INCLUDE_DIR}
     ${PNG_PNG_INCLUDE_DIR}
     ${FREETYPE_INCLUDE_DIRS}
-    ${MIMALLOC_INCLUDE_DIR}
     ${EIGEN3_INCLUDE_DIRS})
 target_link_libraries(slvs_deps INTERFACE
     dxfrw
@@ -35,7 +34,7 @@ target_link_libraries(slvs_deps INTERFACE
     ${ZLIB_LIBRARY}
     ${PNG_LIBRARY}
     ${FREETYPE_LIBRARY}
-    mimalloc-static)
+    mimalloc)
 
 if(NOT USE_QT_GUI OR ENABLE_CLI)
     target_include_directories(slvs_deps SYSTEM INTERFACE ${CAIRO_INCLUDE_DIRS})
--- a/src/slvs/CMakeLists.txt
+++ b/src/slvs/CMakeLists.txt
@@ -5,7 +5,7 @@ add_library(slvs-interface INTERFACE)
 target_sources(slvs-interface INTERFACE lib.cpp)
 target_compile_definitions(slvs-interface INTERFACE -DLIBRARY)
 target_include_directories(slvs-interface INTERFACE ${CMAKE_SOURCE_DIR}/include)
-target_link_libraries(slvs-interface INTERFACE slvs-solver mimalloc-static)
+target_link_libraries(slvs-interface INTERFACE slvs-solver mimalloc)
 
 if(ENABLE_PYTHON_LIB)
     add_custom_command(
--
