diff --git a/GUISupport/Qt/CMakeLists.txt b/GUISupport/Qt/CMakeLists.txt
index c5fc33b..a51ce27 100644
--- a/GUISupport/Qt/CMakeLists.txt
+++ b/GUISupport/Qt/CMakeLists.txt
@@ -191,7 +191,7 @@ if(VTK_BUILD_QT_DESIGNER_PLUGIN AND add_qvtkwidget)
     if(DEFINED VTK_INSTALL_QT_PLUGIN_DIR)
     else()
       set(VTK_INSTALL_QT_PLUGIN_DIR
-        "${DOLLAR}{CMAKE_INSTALL_PREFIX}/${DOLLAR}{VTK_INSTALL_QT_DIR}"
+        "${DOLLAR}{VTK_INSTALL_QT_DIR}"
         CACHE STRING "Directory in which the VTK Qt plugin is placed during installation.")
       mark_as_advanced(VTK_INSTALL_QT_PLUGIN_DIR)
     endif()
