diff --git a/CMake/vtkDirectories.cmake b/CMake/vtkDirectories.cmake
index 3ee98d78..f392cbc6 100644
--- a/CMake/vtkDirectories.cmake
+++ b/CMake/vtkDirectories.cmake
@@ -17,12 +17,12 @@ vtk_set_with_default(VTK_INSTALL_NDK_MODULES_DIR "${VTK_INSTALL_DATA_DIR}/ndk-mo
 # Set up our directory structure for output libraries and binaries
 vtk_set_with_default(CMAKE_RUNTIME_OUTPUT_DIRECTORY "${VTK_BINARY_DIR}/bin")
 if(UNIX)
-  set(vtk_library_directory "lib")
+  set(vtk_library_directory "lib${LIB_SUFFIX}")
 else()
   set(vtk_library_directory "bin")
 endif()
 vtk_set_with_default(CMAKE_LIBRARY_OUTPUT_DIRECTORY "${VTK_BINARY_DIR}/${vtk_library_directory}")
-vtk_set_with_default(CMAKE_ARCHIVE_OUTPUT_DIRECTORY "${VTK_BINARY_DIR}/lib")
+vtk_set_with_default(CMAKE_ARCHIVE_OUTPUT_DIRECTORY "${VTK_BINARY_DIR}/lib${LIB_SUFFIX}")
 vtk_set_with_default(VTK_MODULES_DIR "${VTK_BINARY_DIR}/${VTK_INSTALL_PACKAGE_DIR}/Modules")
 vtk_set_with_default(VTK_WWW_DIR "${VTK_BINARY_DIR}/www")
 
