--- a/GUISupport/Qt/QVTKOpenGLNativeWidget.cxx
+++ b/GUISupport/Qt/QVTKOpenGLNativeWidget.cxx
@@ -21,7 +21,7 @@
 #include <QOpenGLDebugLogger>
 #include <QOpenGLFramebufferObject>
 #include <QOpenGLFunctions>
-#include <QOpenGLFunctions_3_2_Core>
+//#include <QOpenGLFunctions_3_2_Core>
 #include <QOpenGLTexture>
 #include <QPointer>
 #include <QScopedValueRollback>
@@ -532,6 +532,7 @@
   // before proceeding with blit-ing.
   this->makeCurrent();
 
+#if 0
   // blit from this->FBO to QOpenGLWidget's FBO.
   vtkQVTKOpenGLNativeWidgetDebugMacro("paintGL::blit-to-defaultFBO");
   QOpenGLFunctions_3_2_Core* f =
@@ -567,6 +568,7 @@
     f->glColorMask(colorMask[0], colorMask[1], colorMask[2], colorMask[3]);
     f->glClearColor(clearColor[0], clearColor[1], clearColor[2], clearColor[3]);
   }
+#endif  /* 0 */
 }
 
 //-----------------------------------------------------------------------------
