diff -urp oidn-2.0.1.orig/devices/hip/CMakeLists.txt oidn-2.0.1/devices/hip/CMakeLists.txt
--- oidn-2.0.1.orig/devices/hip/CMakeLists.txt	2023-07-15 22:15:48.135408918 -0700
+++ oidn-2.0.1/devices/hip/CMakeLists.txt	2023-07-15 22:18:05.059405376 -0700
@@ -64,7 +64,10 @@ if(WIN32)
   # Fix warning: __declspec attribute 'dllexport' is not supported
   target_compile_options(OpenImageDenoise_device_hip PRIVATE -Wno-ignored-attributes)
 endif()
+
+# oteodoro:  Fixes devices/hip/../../external/composable_kernel/include/ck/utility/math.hpp:171:52: error: no member named 'expf' in namespace 'std'; did you mean simply 'expf'?
+target_compile_options(OpenImageDenoise_device_hip PRIVATE --rocm-path=@ESYSROOT_ROCM_PATH@)    # oteodoro:  set path and use libc++ for distro.
 
 target_link_libraries(OpenImageDenoise_device_hip PRIVATE OpenImageDenoise_core hip::device)
 oidn_strip_symbols(OpenImageDenoise_device_hip)
-oidn_install_module(OpenImageDenoise_device_hip)
\ No newline at end of file
+oidn_install_module(OpenImageDenoise_device_hip)
