--- a/common/cmake/clang.cmake
+++ b/common/cmake/clang.cmake
@@ -23,11 +23,11 @@ IF (EMBREE_ARM)
   ENDIF ()
 ELSE ()
   # for `thread` keyword
-  _SET_IF_EMPTY(FLAGS_SSE2  "-msse -msse2 -mno-sse4.2")
+  _SET_IF_EMPTY(FLAGS_SSE2  "-msse -msse2")
   _SET_IF_EMPTY(FLAGS_SSE42 "-msse4.2")
   _SET_IF_EMPTY(FLAGS_AVX   "-mavx")
   _SET_IF_EMPTY(FLAGS_AVX2  "-mf16c -mavx2 -mfma -mlzcnt -mbmi -mbmi2")
-  _SET_IF_EMPTY(FLAGS_AVX512 "-march=skx")
+  _SET_IF_EMPTY(FLAGS_AVX512 "-mavx512f -mfma -mavx512vl -mavx512bw -mavx512dq")
 ENDIF ()
 
 IF (WIN32)
